src/Hazelcast.Net/HazelcastOptionsBuilder.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using Hazelcast.Exceptions;
+using Microsoft.Extensions.Configuration;
+
+namespace Hazelcast
+{
+    /// <summary>
+    /// Provides a way to build <see cref="HazelcastOptions"/> instances in a fluent way.
+    /// </summary>
+    public class HazelcastOptionsBuilder
+    {
+        private string[] _args;
+        private Dictionary<string, string> _keyValues;
+        private string _optionsFilePath;
+        private string _optionsFileName;
+        private string _environmentName;
+        private List<Action<IConfiguration, HazelcastOptions>> _configureActions;
+
+        /// <summary>
+        /// Sets the command-line arguments to use when building the options.
+        /// </summary>
+        /// <param name="args">The command-line arguments.</param>
+        /// <returns>This options builder.</returns>
+        public HazelcastOptionsBuilder With(string[] args)
+        {
+            _args = args ?? throw new ArgumentNullException(nameof(args));
+            return this;
+        }
+
+        /// <summary>
+        /// Adds a key/value pair to use when building the options.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        /// <param name="value">The value.</param>
+        /// <returns>This options builder.</returns>
+        public HazelcastOptionsBuilder With(string key, string value)
+        {
+            if (string.IsNullOrWhiteSpace(key)) throw new ArgumentException(ExceptionMessages.NullOrEmpty, nameof(key));
+            // assuming value can be null or empty
+
+            _keyValues ??= new Dictionary<string, string>();
+            _keyValues[key] = value;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the path (without filename) to the options files to read when building the options.
+        /// </summary>
+        /// <param name="filePath">The file path.</param>
+        /// <returns>This options builder.</returns>
+        /// <remarks>
+        /// <para>By default, when not provided, the options file is searched in searched in the
```

Comment:
`searched in searched in the` -> `searched in the`

Reply:
ok